src/renderer/mod.rs

```diff
@@ -0,0 +1,244 @@
+    }
+
+    /// Anonymize line numbers
+    ///
+    /// This enables (or disables) line number anonymization. When enabled, line numbers are replaced
+    /// with `LL`.
+    ///
+    /// # Example
+    ///
+    /// ```text
+    ///   --> $DIR/whitespace-trimming.rs:4:193
+    ///    |
+    /// LL | ...                   let _: () = 42;
+    ///    |                                   ^^ expected (), found integer
+    ///    |
+    /// ```
+    pub const fn anonymized_line_numbers(mut self, anonymized_line_numbers: bool) -> Self {
+        self.anonymized_line_numbers = anonymized_line_numbers;
+        self
+    }
+
+    /// Set the margin for the output
+    ///
+    /// This controls the various margins of the output.
+    ///
+    /// # Example
+    ///
+    /// ```text
+    /// error: expected type, found `22`
+    ///   --> examples/footer.rs:29:25
+    ///    |
+    /// 26 | ...         annotations: vec![SourceAnnotation {
+    ///    |                               ---------------- info: while parsing this struct
+    /// ...
+    /// 29 | ...         range: <22, 25>,
+    ///    |                     ^^
+    ///    |
+    /// ```
+    pub const fn margin(mut self, margin: Option<Margin>) -> Self {
+        self.margin = margin;
+        self
+    }
+
+    /// Set the output style for `error`
+    pub const fn error(mut self, style: Style) -> Self {
+        self.stylesheet.error = style;
+        self
+    }
+
+    /// Set the output style for `warning`
+    pub const fn warning(mut self, style: Style) -> Self {
+        self.stylesheet.warning = style;
+        self
+    }
+
+    /// Set the output style for `info`
+    pub const fn info(mut self, style: Style) -> Self {
+        self.stylesheet.info = style;
+        self
+    }
+
+    /// Set the output style for `note`
+    pub const fn note(mut self, style: Style) -> Self {
+        self.stylesheet.note = style;
+        self
+    }
+
+    /// Set the output style for `help`
+    pub const fn help(mut self, style: Style) -> Self {
+        self.stylesheet.help = style;
+        self
+    }
+
+    /// Set the output style for line numbers
+    pub const fn line_no(mut self, style: Style) -> Self {
+        self.stylesheet.line_no = style;
+        self
+    }
+
+    /// Set the output style for emphasis
+    pub const fn emphasis(mut self, style: Style) -> Self {
+        self.stylesheet.emphasis = style;
+        self
+    }
+
+    /// Set the output style for none
+    pub const fn none(mut self, style: Style) -> Self {
+        self.stylesheet.none = style;
+        self
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct StyleSheet {
+    error: Style,
+    warning: Style,
+    info: Style,
+    note: Style,
+    help: Style,
+    line_no: Style,
+    emphasis: Style,
+    none: Style,
+}
+
+impl Default for StyleSheet {
+    fn default() -> Self {
+        Self::const_default()
+    }
+}
+
+impl StyleSheet {
+    pub(crate) const fn const_default() -> Self {
```

Comment:
Why have this separate from `plain`?

When reading the code, its clearer whats intended when looking at the `iml Default` and seeing it call `Self::plain` than `Self::const_default`

Reply:
Whoops, I didn't pay attention to what type this was one.  In this case, I'd rename `const_default` to `plain`